src/js/protocols/webusbdfu.js

```diff
@@ -0,0 +1,1119 @@
+/*
+    WEB USB DFU uses:
+
+    Some references:
+    https://wicg.github.io/webusb/
+    https://developer.mozilla.org/en-US/docs/Web/API/WebUSB_API
+
+    control transfers for communicating
+    recipient is interface
+    request type is class
+
+    General rule to remember is that DFU doesn't like running specific operations while the device isn't in idle state
+    that being said, it seems that certain level of CLRSTATUS is required before running another type of operation for
+    example switching from DNLOAD to UPLOAD, etc, clearning the state so device is in dfuIDLE is highly recommended.
+*/
+import GUI, { TABS } from "../gui";
+import { i18n } from "../localization";
+import { gui_log } from "../gui_log";
+import { usbDevices } from "../usb_devices";
+
+// Task for the brave ones. There are quite a few shadow variables which clash when
+// const or let are used. So need to run thorough tests when chaning `var`
+/* eslint-disable no-var */
```

Comment:
For the new implementation could we avoid old problems? removing this comment ant running `yarn lint` should highlight the issues.

Reply:
Left over from previous refactoring. Removed without issues.